wasmbinding/query_plugin_test.go

```diff
@@ -220,12 +220,60 @@ func (suite *StargateTestSuite) TestConvertProtoToJsonMarshal() {
 func (suite *StargateTestSuite) TestDeterministicJsonMarshal() {
 	testCases := []struct {
 		name                string
+		testSetup           func()
 		originalResponse    string
 		updatedResponse     string
 		queryPath           string
 		responseProtoStruct interface{}
 		expectedProto       func() proto.Message
 	}{
+
+		/**
+		   * Origin Response
+		   * balances:<denom:"bar" amount:"30" > pagination:<next_key:"foo" >
+		   * "0a090a036261721202333012050a03666f6f"
+		   *
+		   * New Version Response
+		   * The binary built from the proto response with additional field address
+		   * balances:<denom:"bar" amount:"30" > pagination:<next_key:"foo" > address:"cosmos1j6j5tsquq2jlw2af7l3xekyaq7zg4l8jsufu78"
+		   * "0a090a036261721202333012050a03666f6f1a2d636f736d6f73316a366a357473717571326a6c77326166376c3378656b796171377a67346c386a737566753738"
+		   // Origin proto
+		   message QueryAllBalancesResponse {
+		  	// balances is the balances of all the coins.
+		  	repeated cosmos.base.v1beta1.Coin balances = 1
+		  	[(gogoproto.nullable) = false, (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"];
+		  	// pagination defines the pagination in the response.
+		  	cosmos.base.query.v1beta1.PageResponse pagination = 2;
+		  }
+		  // Updated proto
+		  message QueryAllBalancesResponse {
+		  	// balances is the balances of all the coins.
+		  	repeated cosmos.base.v1beta1.Coin balances = 1
+		  	[(gogoproto.nullable) = false, (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"];
+		  	// pagination defines the pagination in the response.
+		  	cosmos.base.query.v1beta1.PageResponse pagination = 2;
+		  	// address is the address to query all balances for.
+		  	string address = 3;
+		  }
+		*/
+		{
+			"Query All Balances",
+			func() {
+				wasmbinding.StargateWhitelist.Store("/cosmos.bank.v1beta1.Query/AllBalances", &banktypes.QueryAllBalancesResponse{})
+			},
+			"0a090a036261721202333012050a03666f6f",
+			"0a090a036261721202333012050a03666f6f1a2d636f736d6f73316a366a357473717571326a6c77326166376c3378656b796171377a67346c386a737566753738",
```

Comment:
wait just realized, why is this JSON getting hex encoded?

Reply:
wait no, what is this encoding. Hex decoding gives
```
\n\t\n\x03bar\x12\x0230\x12\x05\n\x03foo\x1a-cosmos1j6j5tsquq2jlw2af7l3xekyaq7zg4l8jsufu78
```
which uhh, does not make any sense to me. Why do we have so many `\x03`, and why do we start `\n\t\n` ?

Reply:
This is a hex encoded version of the response bytes that we would get from the GrpcQueryRouter

Reply:
how did you get `\n\t\n\x03bar\x12\x0230\x12\x05\n\x03foo\x1a-cosmos1j6j5tsquq2jlw2af7l3xekyaq7zg4l8jsufu78`?

Directly hex decoding the given string would give me something like [10 9 10 3 98 97 114 18 2 51 48 18 5 10 3 102 111 111], which if I attempt to convert it to string directly, would give me `bar30 foo` (which is a wrong conversion of string )

Reply:
The following python code:
```
>>> hex_s = "0a090a036261721202333012050a03666f6f1a2d636f736d6f73316a366a357473717571326a6c77326166376c3378656b796171377a67346c386a737566753738"
>>> bytes.fromhex(hex_s)
b'\n\t\n\x03bar\x12\x0230\x12\x05\n\x03foo\x1a-cosmos1j6j5tsquq2jlw2af7l3xekyaq7zg4l8jsufu78'
>>> chr(0x0a) # spot check that the first byte is translated correctly
'\n'
```